src/citrine/informatics/design_spaces/formulation_design_space.py

```diff
@@ -0,0 +1,103 @@
+from typing import Mapping, Optional, Set
+from uuid import UUID
+
+from citrine._rest.resource import Resource
+from citrine._serialization import properties
+from citrine._session import Session
+from citrine.informatics.constraints import Constraint
+from citrine.informatics.descriptors import FormulationDescriptor
+from citrine.informatics.design_spaces.design_space import DesignSpace
+
+__all__ = ['FormulationDesignSpace']
+
+
+class FormulationDesignSpace(Resource['FormulationDesignSpace'], DesignSpace):
+    """[ALPHA] Design space composed of mixtures of ingredients.
+
+    Parameters
+    ----------
+    name: str
+        the name of the design space
+    description: str
+        the description of the design space
+    formulation_descriptor: FormulationDescriptor
+        descriptor used to store formulations sampled from the design space
+    ingredients: Set[str]
+        set of ingredient names that can be used in a formulation
+    constraints: Set[IngredientConstraint]
```

Comment:
This is where I was thinking of elevating a "max_ingredient_count" member, which would result in an automatic IngredientCountConstraint.